src/pow.rs

```diff
@@ -338,16 +340,38 @@ fn count_leading_zeros(data: &[u8]) -> u8 {
 /// ```math
 /// \text{difficulty} = \left(\text{max\_difficulty} - \text{min\_difficulty}\right) \cdot \left(1 - \log_{\text{btc\_per\_emission}}\left(\frac{\text{balance} - \text{min\_balance}}{\text{min\_balance}}\right)\right)+\text{min\_difficulty}
 /// ```
+///
+/// # Errors
+///
+/// This function will return an error if any of the following conditions are met:
+///
+/// - `min_balance` is zero
+/// - `max_difficulty` is less than `min_difficulty`
+/// - `per_emission` is zero or equal to 1 BTC
 pub fn calculate_difficulty(
     max_difficulty: f32,
     min_difficulty: f32,
     balance: Amount,
     min_balance: Amount,
     per_emission: Amount,
-) -> f32 {
+) -> Result<f32, DifficultyCalculationError> {
+    if min_balance == Amount::ZERO {
+        return Err(DifficultyCalculationError::MinBalanceZero);
+    }
```

Comment:
To be honest this should not be a `DifficultyCalculationError` but more like a big fucking error. If comes to this, we're toasted... This should be another type of error and we convert maybe?

But it is fine for now.